main.go

```diff
@@ -157,7 +155,7 @@ func main() {
 	log.FromContext(ctx).Infof("executing phase 3: parsing network prefixes for ipam")
 	// ********************************************************************************
 
-	ipamChain := getIPAMChain(ctx, cfg.CidrPrefix)
+	ipamChain := groupipam.NewServer(cfg.CidrPrefix)
```

Comment:
```suggestion
```

Reply:
Please do it inline at 170 

Also note, that here we should use `singlepointipam`.

Consider to use option groupipam.WithCustomImapServer(singlepointipam.NewServer)